### PR TITLE
tools/stellar-sign add verify feature

### PR DESCRIPTION
**Pull request by C0x41lch0x41**
- [tools/stellar-sign] Added capability to verify a TX vs signing it
- [tools/stellar-sign] Update README

<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Added the possibility to verify a transaction signature in stellar-sign

### Why

I needed a quick standalone tool to verify transactions signatures. 
I could not find it and stellar-sign seemed like the right place to add this.

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: tools/stellar-sign/README.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
 # Stellar Sign
 
-This folder contains `stellar-sign` a simple utility to make it easy to add your signature to a transaction envelope.  When run on the terminal it:
+This folder contains `stellar-sign` a simple utility to make it easy to add your signature to a transaction envelope or to verify a transactio signature with a public key.
```
Comment (jacekn) [inline tools/stellar-sign/README.md:3]:
```suggestion
This folder contains `stellar-sign` a simple utility to make it easy to add your signature to a transaction envelope or to verify a transaction signature with a public key.  
```

File: tools/stellar-sign/main.go [hunk 2]
```diff
@@ -1,28 +1,95 @@
...
+		}
+	}
+	return newEnv, err
+}
+
 var in *bufio.Reader
 
 var infile = flag.String("infile", "", "transaction envelope")
+var verify = flag.Bool("verify", false, "Verify the transaction instead of signing")
+var testnet = flag.Bool("test", false, "Testnet")
```
Comment (bartekn) [inline tools/stellar-sign/main.go:92]: I think `test` can be confusing. Maybe just use `testnet`?
```suggestion
var testnet = flag.Bool("testnet", false, "Verifies or signs tx using testnet passphrase")
```

File: tools/stellar-sign/main.go [hunk 3]
```diff
@@ -70,57 +137,48 @@ func main() {
...
-		tx, err = tx.Sign(network.PublicNetworkPassphrase, kp)
-		if err != nil {
-			log.Fatal(err)
-		}
-		newEnv, err = tx.Base64()
+	passPhrase := network.PublicNetworkPassphrase
+	if *testnet {
+		passPhrase = network.TestNetworkPassphrase
+	}
+	newEnv, err := flowRouter.do(parsed, passPhrase)
```
Comment (bartekn) [inline tools/stellar-sign/main.go:168]: Nit: I understand a need for extracting the code to a separate struct. However, I'm not sure if both functions should be done inside `do` method - verify part doesn't modify tx so not sure why it returns `newEnv`. Why not simply add `Sign` and `Verify` methods to `SignOrVerify`?

**Comment by C0x41lch0x41** [comment]
@bartekn thanks for the comments. I updated the code, please let me know what you think. Thanks!

**Comment by C0x41lch0x41** [comment]
@bartekn thanks for the comments. I updated the code, please let me know what you think. Thanks!

**Review by bartekn — APPROVED** [review]

**Pull request merged**